var/spack/repos/builtin/packages/py-pillow/package.py

```diff
@@ -77,32 +78,48 @@ def patch(self):
         include_dirs = []
         for dep in self.spec.dependencies(deptype='link'):
             query = self.spec[dep.name]
-            library_dirs.extend(query.libs.directories)
-            include_dirs.extend(query.headers.directories)
+            if 'lcms' not in query:
+                library_dirs.extend(query.libs.directories)
+                include_dirs.extend(query.headers.directories)
 
         setup = FileFilter('setup.py')
+        if '+lcms' in self.spec:
+            setup.filter('LCMS_ROOT = None',
+                         'LCMS_ROOT = ("{0}", "{1}")'.format(
+                             self.spec['lcms'].prefix.lib,
+                             self.spec['lcms'].prefix.include))
+
         setup.filter('library_dirs = []',
                      'library_dirs = {0}'.format(library_dirs), string=True)
         setup.filter('include_dirs = []',
                      'include_dirs = {0}'.format(include_dirs), string=True)
-
-    def build_ext_args(self, spec, prefix):
-        def variant_to_flag(variant):
-            able = 'enable' if '+' + variant in spec else 'disable'
-            return '--{0}-{1}'.format(able, variant)
-
-        args = ['--enable-zlib', '--enable-jpeg']
-
-        variants = ['tiff', 'freetype', 'lcms', 'webp', 'webpmux', 'jpeg2000']
-        args.extend(list(map(variant_to_flag, variants)))
-
-        # Spack does not (yet) support these modes of building
-        args.append('--disable-imagequant')
-
-        args.append('--rpath={0}'.format(':'.join(self.rpath)))
-        return args
-
-    # Tests need to be re-added since `phases` was overridden
-    run_after('build_ext')(PythonPackage.test)
-    run_after('install')(PythonPackage.import_module_test)
-    run_after('install')(PythonPackage.sanity_check_prefix)
+        self._pillow_setup()
+
+    def _pillow_setup(self):
+
+        def variant_to_cfg(setup):
+            able = 'enable' if '+' + variant in self.spec else 'disable'
+            return '{0}-{1}=1\n'.format(able, variant)
+
+        with open('setup.cfg', 'a') as setup:
+            # Default backend
+            setup.write('[build_ext]\n')
+            setup.write('enable-zlib=1\n')
+            setup.write('enable-jpeg=1\n')
+            setup.write('debug=1\n')
```

Comment:
This wasn't present before, why the change? Maybe this could be a variant.

Reply:
Oops, that should be removed.